kvbc/src/categorization/kv_blockchain.cpp

```diff
@@ -258,6 +266,9 @@ bool KeyValueBlockchain::deleteBlock(const BlockId& block_id) {
     return false;
   }
 
+  // Lets update the delete metrics component
+  delete_metrics_comp_.UpdateAggregator();
```

Comment:
Why are we updating the aggregator on every single block deletion? Wouldn't it be better to update after pruning is completed, or at least intermittently, every 1000 or so blocks?

Reply:
We call deleteBlock only on pruning (for now at least). Updating the aggregator has almost zero effect on pruning performance (in the worse case we will wait for a lock that is taken by the puller of the aggregator).
Thus, in this specific case, I would prefer a better granularity of the metrics more than a very slightly performance improvement (if any).
WDYT?

Reply:
I think it's a waste of resources, but probably not that important right now.I'm fine leaving it as is for now. We can always change it later and I suspect we will, but maybe not until we allow pruning to interleave with normal operation in which case the lock contention/cache coherency will matter more.